src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -126,48 +126,45 @@ private void backup(final JavaFile javaFile) throws IOException {
-    // TODO: for cleaners, santa clauses, etc.
-    private void applyFix(final String polluterName,
-                          final JavaMethod cleanerMethod,
-                          final JavaMethod victimMethod) throws Exception {
-        backup(victimMethod.javaFile());
-
-        // Compile the test without our fix
-        runMvnInstall();
-
-        // Check if we pass in isolation before fix
-        TestPluginPlugin.info("Running victim test with polluter before adding code from cleaner.");
-        final List<String> withPolluter =
-                // This is to handle the case of a santa clause, where a test will fail with no polluter
-                polluterName.isEmpty() ? ListUtil.fromArray(victimMethod.methodName()) :
-                                         ListUtil.fromArray(polluterName, victimMethod.methodName());
-        if (testOrderPasses(withPolluter)) {
-            TestPluginPlugin.error("Test passes with the polluter, but is supposed to fail.");
-            return;
-        }
-
-        // Do our fix
-        TestPluginPlugin.info("Applying code from cleaner and recompiling.");
-        final NodeList<Statement> cleanerStmts = cleanerMethod.body().getStatements();
-
+    // Try applying cleaner statements and see if test passes when run after polluter
+    // Returns true if passes, returns false if not
+    private boolean checkCleanerStmts(final String polluterName,
+                                      final JavaMethod victimMethod,
+                                      final NodeList<Statement> cleanerStmts) throws Exception {
         // Note: this modifies victimMethod, so when we eventually make it delta-debug, we will want
         // to make a copy of the victimMethod somewhere
         victimMethod.prepend(cleanerStmts);
         victimMethod.javaFile().writeAndReloadCompilationUnit();
 
-        // Recompile and check if we pass in isolation after fix
-        runMvnInstall();
-
+        // Rebuild and see if tests run properly
+        try {
+            runMvnInstall();
+        } catch (Exception ex) {
+            TestPluginPlugin.info("Error building the code, passed in cleaner does not work");
+            // Reset the change
+            victimMethod.removeFirstBlock();
+            return false;
+        }
         // TODO: Output to result files rather than stdout
-        TestPluginPlugin.info("Running victim test in isolation with code from cleaner.");
-        final boolean passInIsolationAfterFix = testOrderPasses(withPolluter);
+        TestPluginPlugin.info("Running victim test with code from cleaner.");
+        List<String> tests;
+        if (polluterName.isEmpty()) {
+            tests = ListUtil.fromArray(victimMethod.methodName());
+        } else {
+            tests = ListUtil.fromArray(polluterName, victimMethod.methodName());
+        }
+        boolean passInIsolationAfterFix = testOrderPasses(tests);
```

Comment:
Minor: `passInIsolationAfterFix` should probably be renamed, given that `tests` might include the polluter.